### ISSUE TITLE
Object type checking might fail in nuxt applications

### ISSUE THREAD

**Issue by LucidityDesign** [wopian/kitsu#652]
For response objects coming from nuxt-axios the test for `obj.constructor === Object` (as it is used [here](https://github.com/wopian/kitsu/blob/a769ac5d8cfd2b3fdb6c98a5ceb6ba1d109e244a/packages/kitsu-core/src/deattribute/index.js#L32)) fails.

Also see https://github.com/nuxt-community/axios-module/issues/565

Can we use a more robust logic to check for objects?

**Comment by wopian** [comment]
Does `obj instanceof Object` work for `nuxt-axios`?

**Comment by LucidityDesign** [comment]
I tried several checks I found on StackOverflow:
```
test.data instanceof Object => false
typeof test.data => "object"
test.data.constructor.name => "Object"
Object.prototype.toString.call(test.data) => "[object Object]"
Object.getPrototypeOf(test.data) === Object.prototype => false
```
I will have another look at the axios module.
I can start the demo with OpenSSL 3.0.1

**Comment by wopian** [comment]
I'll switch them to `typeof` 👍

**Comment by LucidityDesign** [comment]
Thanks. Keep in mind, that `null` and Arrays also return `'object'`

**Comment by wopian** [comment]
Released as 9.1.25 and 10.0.0-alpha.22
